src/editor.js

```diff
@@ -1257,6 +1296,124 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    // PASTING / DROPPING
+
+    /**
+     * Prepare clipboard data (text/html) for safe pasting into the editor.
+     *
+     * @private
+     * @param {string} clipboardData
+     * @returns {string}
+     */
+     _prepareClipboardData(clipboardData) {
+        const container = document.createElement('fake-container');
+        container.innerHTML = clipboardData;
+        return this._cleanForPaste(container).innerHTML;
+    }
+    /**
+     * Prepare clipboard data (text/plain) for safe pasting into the editor.
+     *
+     * @private
+     * @param {string} clipboardData
+     * @returns {string}
+     */
+    _prepareTextClipboardData(clipboardData) {
+        const isXML = !!clipboardData.match(/<[a-z]+[a-z0-9-]*( [^>]*)*>[\s\S\n\r]*<\/[a-z]+[a-z0-9-]*>/i);
+        const isJS = !isXML && !!clipboardData.match(/\(\);|this\.|self\.|function\s?\(|super\.|[a-z0-9]\.[a-z].*;/i);
+
+        const container = document.createElement('fake-container');
+        const pre = document.createElement('pre');
+        pre.innerHTML = clipboardData.trim()
+            .replace(/</g, '&lt;').replace(/>/g, '&gt;')
+            // Get that text as an array of text nodes separated by <br> where
+            // needed.
+            .replace(/(\n+)/g, '<br>');
+
+        if (isJS || isXML) {
+            container.appendChild(pre);
+        } else {
+            for (const node of pre.childNodes) {
+                container.appendChild(node);
+            };
+        }
+        return container.innerHTML;
+    }
+    /**
+     * Clean a container element's children (or the container itself if
+     * `justContainer` is true) for safely pasting. Cleaning an element involves
+     * unwrapping its contents if it's an illegal (blacklisted or not
+     * whitelisted) element, or removing its illegal attributes and classes.
+     * Return the cleaned element.
+     *
+     * @param {Node} container
+     * @param {boolean} [justContainer=false]
+     * @returns {Node}
+     */
+    _cleanForPaste(container, justContainer = false) {
+        for (const node of justContainer ? [container] : [...container.childNodes]) {
+            if (!this._isWhitelisted(node) || this._isBlacklisted(node)) {
+                if (['META', 'STYLE', 'SCRIPT'].includes(node.nodeName)) {
```

Comment:
I don't understand what you mean. However, I'm in favor of having blacklist as an object if only for the sake of consistency.

Reply:
Here I'm repeating some of the blacklisted selectors because these in particular I don't even want to keep their children, I want to nuke them. So I thought I could maybe also have two keys in my blacklist so I could identify these.
(Oh and that's funny cause it also reminded me of that debate :p If you think back on it though you'll recall that while I disagreed with you on the case of `master`, I didn't regarding `whitelist` and `blacklist` ^^ - though I'm not excluding that there's history I'm not aware of there).

Reply:
I was mostly trolling... :p I think your recollection of the debate is accurate though, my bad. I would advise for something else than `nuke` though. Probably `remove` ?

Reply:
Sure I just played along on the troll, no worries :-P
`remove` sounds good :+1: